ingress: Update docs with network policy example (#31060)
cilium-team/cilium: Documentation/network/servicemesh/http.rst

      "ISBN-10": "1234567890",
      "ISBN-13": "123-1234567890"
    }

Apply Cilium Network Policy
===========================

> The above example doesn't include any network policy, so all the external traffic is allowed by default.

Reviewer: "Above" in this context sometimes trips up translators. Can we change this to 'previous'?

```suggestion
The previous example doesn't include any network policy, so all the external traffic is allowed by default.
```

> Let's apply network policy to lock down the external traffics.

Reviewer: Typo? 

```suggestion
Let's apply network policy to lock down the external traffic.
```

.. literalinclude:: ../../../examples/kubernetes/servicemesh/policy/external-lockdown.yaml

.. parsed-literal::

    $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/servicemesh/policy/external-lockdown.yaml

> The same request should be rejected with 503 Service Unavailable status code for any request originating from outside.

Reviewer: Did you mean `403` here instead of `503`? Also small wording and grammar suggestion:

```suggestion
The same request from the previous example should now be rejected with a 403 Service Unavailable status code:
```

Another option to make the statement more generic:

```suggestion
With this policy applied, any request originating from outside the cluster will be rejected with a 403 Service Unavailable status code:
```

    *   Trying 172.18.255.194:80...
    * Connected to 172.18.255.194 (172.18.255.194) port 80
    > GET /details/1 HTTP/1.1
    > Host: 172.18.255.194
    > User-Agent: curl/8.6.0
    > Accept: */*
    >
    < HTTP/1.1 403 Forbidden
    < content-length: 15
    < content-type: text/plain
    < date: Thu, 29 Feb 2024 12:59:54 GMT
    < server: envoy
    * The requested URL returned error: 403
    * Closing connection
    curl: (22) The requested URL returned error: 403

    # Capture hubble flows in another terminal
    $ kubectl --namespace=kube-system exec -i -t cilium-xjl4x -- hubble observe -f --identity ingress
    Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), wait-for-node-init (init), clean-cilium-state (init), install-cni-binaries (init)
    Feb 29 13:00:29.389: 172.18.0.1:53866 (ingress) -> kube-system/cilium-ingress:80 (world) http-request DROPPED (HTTP/1.1 GET http://172.18.255.194/details/1)
    Feb 29 13:00:29.389: 172.18.0.1:53866 (ingress) <- kube-system/cilium-ingress:80 (world) http-response FORWARDED (HTTP/1.1 403 0ms (GET http://172.18.255.194/details/1))

> Let's check if the in-cluster traffic to Ingress endpoint is still allowed.

Reviewer: ```suggestion
Let's check if the in-cluster traffic to Ingress endpoint is still allowed:
```
